CMakeLists.txt

```diff
@@ -251,9 +254,12 @@ if(W MATCHES "3")
   toolchain_cc_warning_dw_3()
 endif()
 
-# @Intent Add extended, more specific, toolchain warning flags
+# @Intent: Add extended, more specific, toolchain warning flags
 toolchain_cc_warning_extended()
 
+# @Intent: Add toolchain flags for turning warnings into errors
+toolchain_cc_warning_error()
```

Comment:
That's not what -Werror=implicit-int does.

Reply:
The -Werror=implicit-int turns the -Wimplicit-int warning into an error, hence the -Werror=.
As the commit message states, the macro is intended as a common placeholder for all, present and future, -Werror flags, thus collecting those in a common location so that other toolchains may add and/or remove this kind of flags as needed.

Reply:
"Trigger errors for a set of warnings"

is too high-level of an abstraction.

Imagine that in the future we want to, for certain configurations,
enable certain errors, and disable others.

With this abstraction we would not be able to do this.

It would be more maintainable if we had an intent per-flag, e.g.:

"Trigger an error when a declaration does not specify a type."

Reply:
May I then suggest the compromise: `toolchain_cc_implicit_int_error` whoose intent is to obtain the flag that turns implicit ints into errors.

Reply:
This would work.

Reply:
@SebastianBoe A high level of abstraction is what makes it scale well between different toolchains making it very easy to add and remove flags per tool chain, and having all flags related to warnings and errors in one common location makes it easy to maintain those types of flags which in turn would help to make Zephyr more portable, wouldn't you say?

Of cause as @mped-oticon states, this flag could be obtained by a macro, just like the misra sane flags, but that means that the macro will be locked to that flag only, thus making it impossible to add other flags for other toolchains. 
As for select/deselect of flags based upon configuration, this would be possible in the toolchain_cc_warning_error macro, in the same way as it is done in the CMakeLists.txt file today. Actually, the misra sane flags could have been moved entirely to the macro, thus keeping the CMakeLists.txt file simple and easy to maintain. 
Example: 
```
macro(toolchain_cc_warning_error)

  if(CONFIG_MISRA_SANE)
    zephyr_compile_options($<$<COMPILE_LANGUAGE:C>:-Werror=vla>)
    zephyr_compile_options($<$<COMPILE_LANGUAGE:CXX>:-Werror=vla>)
  endif()

  # Force an error when things like SYS_INIT(foo, ...) occur with a missing header
  zephyr_cc_option(-Werror=implicit-int)

endmacro()
```

Perhaps you could tell me if there is some deeper meaning as to why the above macro would not be simpler than having many separate macros and thus places to maintain those types of flags?

Reply:
You are correct that it is easier to introduce new flags if we have fewer abstractions, as one would only need to add the flag to each toolchain implementation, instead of creating a new abstraction, and also adding the flag to each toolchain implementation.

But it does not scale well due to how other scenario's play out.

Imagine that at first, we always apply a flag, but then later on find out that this flag is undesirable in certain situations. If we had an abstraction for this flag we could just do;

if(CONFIG_NO_FLAG_X)
toolchain_cc_do_flag_x()
endif()

But without one we need to first go into each toolchain (out-of-tree toolchains being inaccessible, would also mean they would need to break), and create this abstraction for each of them, and then introduce the corresponding change to the toplevel CMakeLists.txt.

Reply:
Ok @SebastianBoe, thank you for the conversation. 

I see your point about having to maintain several files, but in the scenario you mention, it would only be the toolchain placeholder macro(s) pertaining to the undesired flag(s) which would need to be updated and no change would be needed in the root CMakeLists.txt file. As for out-of-tree toolchains, they might not need that change and if they do, it should be up to the user of the out-of-tree toolchain and not the Zephyr project to maintain that, since as you correctly state, "out-of-tree toolchains being inaccessible".

In any case, I will change the Intent and the macro to handle that one flag only. Thank you again for the dialog.